src/js/parsers/captions/dfxp.ts

```diff
@@ -6,9 +6,10 @@ import type { CaptionEntryData } from './captions.types';
 
 type ParagraphNodes = HTMLCollectionOf<Element & { text?: string }>;
 
-export default function Dfxp(xmlDoc: XMLDocument): CaptionEntryData[] {
+export default function Dfxp(xmlDoc: XMLDocument | null): CaptionEntryData[] {
     if (!xmlDoc) {
         parseError(306007);
+        return [];
```

Comment:
`parseError` throws so we do not need to return here. Try using `never` below:

```ts
function parseError(code: number): never {
```

Reply:
can I return parseError then? Need to pass type-check

Reply:
`never` indicates that the function will not complete.

Reply:
nice, didn't see the screenshot before for some reason. Trying it out now!